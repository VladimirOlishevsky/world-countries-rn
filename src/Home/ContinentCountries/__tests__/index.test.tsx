import React from "react";
import { render } from '@testing-library/react-native';
import { ContinentCountries } from "..";
import { enableFetchMocks } from 'jest-fetch-mock';

enableFetchMocks();

const mockFetchRegion = jest.fn();
const mockStore = {
  continentsStore: {
    fetchRegions: mockFetchRegion,
    countries: [
      {
        Name: 'South',
        FlagPng: 'south flag',
        Alpha2Code: 'so'
      },
      {
        Name: 'North',
        FlagPng: 'north flag',
        Alpha2Code: 'no'
      }
    ]
  },

};

jest.mock('store/index', () => ({
  getRootStore: () => mockStore,
}));


describe('ContinentCountries', () => {
  it('render component ContinentCountries', async () => {
    const route = {
      params: {
        name: 'Some country'
      }
    }
    const component = await render(<ContinentCountries route={route} />);
    expect(component).toMatchSnapshot();
  });
  //   it('component ContinentCards action fetchRegions', async () => {

  //     const { getByTestId } = render(<ContinentCards />);

  //     const page = getByTestId('Europe').props as {
  //       onClick: () => void
  //     };;

  //     expect(mockFetchProfileType).not.toBeCalled();
  //     page.onClick()
  //     expect(mockFetchProfileType).toBeCalledTimes(1);
  //   });
});