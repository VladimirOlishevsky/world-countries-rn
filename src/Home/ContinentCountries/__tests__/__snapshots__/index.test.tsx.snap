// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContinentCountries render component ContinentCountries 1`] = `
<View
  style={
    Object {
      "backgroundColor": undefined,
      "height": 1334,
      "padding": NaN,
      "paddingBottom": NaN,
      "paddingTop": null,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": undefined,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": NaN,
          "justifyContent": "center",
          "opacity": 1,
          "width": NaN,
        }
      }
    >
      <Icon
        color="#ff9f0a"
        name="chevron-left"
      />
    </View>
    <Text
      style={
        Object {
          "color": undefined,
          "fontSize": undefined,
          "marginLeft": NaN,
        }
      }
    >
      Some country
    </Text>
  </View>
  <RCTScrollView>
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": NaN,
            "marginTop": NaN,
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "south flag",
            }
          }
          style={
            Object {
              "height": NaN,
              "width": NaN,
            }
          }
        />
        <Text
          style={
            Object {
              "color": undefined,
              "fontSize": undefined,
              "marginLeft": undefined,
            }
          }
        >
          South
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": NaN,
            "marginTop": NaN,
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "north flag",
            }
          }
          style={
            Object {
              "height": NaN,
              "width": NaN,
            }
          }
        />
        <Text
          style={
            Object {
              "color": undefined,
              "fontSize": undefined,
              "marginLeft": undefined,
            }
          }
        >
          North
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
